downstreamadapter/dispatchermanager/helper.go

```diff
@@ -173,6 +173,19 @@ func newSchedulerDispatcherRequestDynamicStream() dynstream.DynamicStream[int, c
 	return ds
 }
 
+func schedulerDispatcherKey(req *heartbeatpb.ScheduleDispatcherRequest) string {
+	if req == nil || req.Config == nil {
+		return ""
+	}
+	if req.Config.Span != nil {
```

Comment:
why sometimes return span's string, sometimes return dispatcherID?

Reply:
There's a special case where the maintainer finds that the dispatcher manager reported a working dispatcher, but the span controller doesn't have a corresponding span. The maintainer will then send a remove operator request to remove this dispatcher. Since the span doesn't exist, the request won't include span information; the dispatcher ID will be used instead. However, there won't be a conflict because the span ID and dispatcher ID will never be the same.